demos/sev-demo/README.md

```diff
@@ -0,0 +1,136 @@
+
+You can run `gop-server.py` wherever you would like, just make sure that `guest_attestation_proxy` is set accordingly.
+
+The KBS is configured using two files. First, `keys.json` stores the keys that the KBS can send to a guest. Generating these keys is described below. There is also a file called `keysets.json` which defines groups of keys and the policies that correspond to their release. Since the `offline_sev_kbc` is an offline KBC, keys cannot be request individually at runtime. Instead, all the keys that are needed for the duration of the guest need to be provided at boot. Thus, we use a keyset to group keys into one bundle that is injected. 
+
+We can step through the keysets file.
+```
+{
+   "KEYSET-1":{
+```
+Currently we are hardcoding our keyset in the Kata configuration file (see above) to match `KEYSET-1` specified here. 
+```
+      "allowed_digests":[
+         "a11cc9f32b3e64a0846b5eadf773e5e4ee7b4187d984b7a72a3b4fc99e34374c",
+         "dcf332c8abb68cc534d7f1e523ffe9e072308e0d92dc98e3884b313caaea379b"
+      ],
+```
+Firmware digests that are allowed for this keyset. You can calculate a firmware digest using the `calculate_hash.py` tool. Be sure to specify the correct, kernel, initrd, firmware, and kernel parameters. These will be included in the hash (this is an extension of the original SEV behavior). You should have the path to these components in your kata configuration file. Do not use the kernel parameters field from the configuration file, though, because it only contains additional parameters rather than he full command line. The easiest way to get the full kernel parameters is to start a container with Kata and then check the QEMU arguments via `journalctl -xe -u containerd` or `dmesg | grep qemu`. Note that viewing the kernel commandline from inside the guest via `cat /proc/cmdline` might not accurately reflect the firmware's view of the kernel commandline, which is what the measurement is based on.  
+```
+      "allowed_keys":[
+         "key_id1"
+      ],
+```
+This should correspond to the key in `keys.json`.
+```
+      "allowed-policies":[
+         0
+      ],
+      "min-api-major":0,
+      "min-api-minor":0,
```

Comment:
Why don't we just specify a [semver](https://semver.org/) (or partial semver) here (as a single value)? Something like:

```json
"min-api-version": "1.2",
```
or,

```json
"min-api-version": "1.2.3",
```


This comment does assume that QEMU's QMP `query-version` command is semver compatible. I'm not sure, but it looks like it is:

- https://wiki.qemu.org/Features/QAPI#Complex_Types

Reply:
This is the PSP firmware version not the QEMU version. It would probably be a good idea to combine those fields, though.